cmd/oras/root/manifest/index/create.go

```diff
@@ -0,0 +1,207 @@
+	return pushIndex(ctx, target, desc, content, opts.Reference, opts.extraRefs, opts.AnnotatedReference(), opts.Printer)
+}
+
+func resolveSourceManifests(ctx context.Context, target oras.ReadOnlyTarget, opts createOptions) ([]ocispec.Descriptor, error) {
+	var resolved []ocispec.Descriptor
+	for _, source := range opts.sources {
+		desc, content, err := oras.FetchBytes(ctx, target, source, oras.DefaultFetchBytesOptions)
+		if err != nil {
+			if errors.Is(err, errdef.ErrNotFound) {
+				return nil, fmt.Errorf("could not resolve %s. Make sure the manifest exists in %s", source, opts.Path)
+			}
+			return nil, err
+		}
+		opts.Println("Resolved manifest", source)
+		if descriptor.IsImageManifest(desc) {
+			desc.Platform, err = getPlatform(ctx, target, content)
+			if err != nil {
+				return nil, err
+			}
+		}
+		resolved = append(resolved, desc)
+	}
+	return resolved, nil
+}
+
+func getPlatform(ctx context.Context, target oras.ReadOnlyTarget, manifestBytes []byte) (*ocispec.Platform, error) {
+	// extract config descriptor
+	var manifest ocispec.Manifest
+	if err := json.Unmarshal(manifestBytes, &manifest); err != nil {
+		return nil, err
+	}
+	// fetch config content
+	contentBytes, err := content.FetchAll(ctx, target, manifest.Config)
+	if err != nil {
+		return nil, err
+	}
+	var platform ocispec.Platform
+	if err := json.Unmarshal(contentBytes, &platform); err != nil {
+		return nil, err
+	}
+	return &platform, nil
+}
+
+func packIndex(oldIndex *ocispec.Index, manifests []ocispec.Descriptor) (ocispec.Descriptor, []byte, error) {
+	index := ocispec.Index{
+		Versioned: specs.Versioned{
+			SchemaVersion: 2,
+		},
+		MediaType:    ocispec.MediaTypeImageIndex,
+		ArtifactType: oldIndex.ArtifactType,
+		Manifests:    manifests,
+		Subject:      oldIndex.Subject,
+		Annotations:  oldIndex.Annotations,
+	}
+	indexBytes, err := json.Marshal(index)
+	if err != nil {
+		return ocispec.Descriptor{}, nil, err
+	}
+	desc := content.NewDescriptorFromBytes(ocispec.MediaTypeImageIndex, indexBytes)
+	return desc, indexBytes, nil
+}
+
+func pushIndex(ctx context.Context, target oras.Target, desc ocispec.Descriptor, content []byte, ref string, extraRefs []string, path string, printer *output.Printer) error {
+	// push the index
+	var err error
+	if ref == "" || contentutil.IsDigest(ref) {
+		err = target.Push(ctx, desc, bytes.NewReader(content))
+	} else {
+		_, err = oras.TagBytes(ctx, target, desc.MediaType, content, ref)
+	}
+	if err != nil {
+		return err
+	}
+	printer.Println("Pushed", path)
+
+	// tag the index if extra tags are provided
+	extraRefs = removeDigests(extraRefs)
+	if len(extraRefs) != 0 {
+		handler := display.NewManifestPushHandler(printer)
+		tagListener := listener.NewTaggedListener(target, handler.OnTagged)
+		if _, err = oras.TagBytesN(ctx, tagListener, desc.MediaType, content, extraRefs, oras.DefaultTagBytesNOptions); err != nil {
+			return err
+		}
+	}
+	return printer.Println("Digest:", desc.Digest)
```

Comment:
Looks like this should be output before tagging.

Reply:
Has a point. If this is printed before tagging and tagging fails, user knows what's the digest. But I wrote it to align with existing status output of `oras manifest push`.